android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java

```diff
@@ -62,28 +63,45 @@ public void onClick(View v) {
                 password = passwordEditText.getText().toString();
 
                 LoginRequest loginRequest = new LoginRequest(username, password);
-                Call<SingleStringResponse> loginQuery = retrofitInterface.login(loginRequest);
+                Call<SingleResponse<String>> loginQuery = retrofitInterface.login(loginRequest);
 
-                loginQuery.enqueue(new Callback<SingleStringResponse>() {
+                loginQuery.enqueue(new Callback<SingleResponse<String>>() {
                     @Override
-                    public void onResponse(Call<SingleStringResponse> call, Response<SingleStringResponse> response) {
+                    public void onResponse(Call<SingleResponse<String>> call, Response<SingleResponse<String>> response) {
                         if (response.isSuccessful() && response.body() != null && response.body().hasResponse()) {
-                            String token = response.body().getResponse();
+                            String token = response.body().getData();
 
                             // TODO: Store the token properly here
                             tokenHandler.setParentToken(token);
 
-                            // TODO: Navigate to profile page
+                            Call<AccountResponse> accountQuery = retrofitInterface.me(tokenHandler.getToken());
+
+                            accountQuery.enqueue(new Callback<AccountResponse>() {
+                                @Override
+                                public void onResponse(Call<AccountResponse> call, Response<AccountResponse> response) {
+                                    // TODO: Navigate to profile page
+                                    navigateTo(ParentHomeActivity.class,
+                                            getResources().getString(R.string.clan_name),
+                                            response.body().getData().getClanName());
```

Comment:
check if response is valid

Reply:
I just tested the scenario of the returned schema changing to simulate invalid response and the `onFailure` handler will get called in that case.